File: src/main/java/br/com/rotaractsorocabaleste/rotasales/event/EventServiceImpl.java
package br.com.rotaractsorocabaleste.rotasales.event;

import br.com.rotaractsorocabaleste.rotasales.user.UserService;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.Optional;
import java.util.Set;
import java.util.UUID;

@Service
@RequiredArgsConstructor
public class EventServiceImpl implements EventService {

    private final EventRepository eventRepository;
    private final UserService userService;

    @Override
    public Event create(final Event event) {
        return eventRepository.save(event);
    }

    @Override
    public Optional<Event> findById(final UUID id) {
        return eventRepository.findById(id);
    }

    @Override
    public boolean existsById(final UUID id) {
        return eventRepository.existsById(id);
    }

    @Override
    public Set<Event> findMyInstitutionActiveEvents() {
        final var loggedInUser = userService.findLoggedInUser();
        final var institutionId = loggedInUser.getInstitution().getId();

        return eventRepository.findByInstitutionIdAndActive(institutionId);
    }

}

File: src/main/kotlin/br/com/rotaractsorocabaleste/rotasales/core/service/impl/EventServiceImpl.kt
package br.com.rotaractsorocabaleste.rotasales.core.service.impl

import br.com.rotaractsorocabaleste.rotasales.core.entity.Event
import br.com.rotaractsorocabaleste.rotasales.core.service.EventService
import br.com.rotaractsorocabaleste.rotasales.dataprovider.repository.EventRepository
import org.springframework.stereotype.Service
import java.time.ZonedDateTime

@Service
class EventServiceImpl(
    private val eventRepository: EventRepository
) : EventService {

    override fun getActiveEvents(): List<Event> {
        return eventRepository
            .findByEndDateGreaterThanEqual(
                ZonedDateTime
                    .now()
                    .withHour(0)
                    .withMinute(0)
                    .withSecond(0)
            )
    }

}